Cannot build from source

Thanks for putting little-buddy together!  I'm using it now, and it is working great.

I'm also interested in contributing, but when I try to build, I get some errors https://gist.github.com/ryantrinkle/8e8df486fb8bf0296f82fb8ecdba80f1

Let me know if there's anything I can do to help narrow this down!

I'm on `x86_64-linux` as well. What about `nix develop` then `make`? Here are the build logs I get:

```
❯ make
MAKE START: 2024-01-20 22:41:12.181436990
make -C /home/bryton/src/github.com/hall/little-buddy/out KBUILD_SRC=/home/bryton/src/github.com/hall/little-buddy \
        -f /home/bryton/src/github.com/hall/little-buddy/Makefile 
make[1]: Entering directory '/home/bryton/src/github.com/hall/little-buddy/out'
-------------------------------
REVISION_INFO: 1e9b4f1
-------------------------------
/home/bryton/src/github.com/hall/little-buddy/Makefile:583: warning: overriding recipe for target 'clean'
/home/bryton/src/github.com/hall/little-buddy/Makefile:574: warning: ignoring old recipe for target 'clean'
make -f ../scripts/build.mk obj=platform 
make -f ../scripts/build.mk obj=platform/cmsis 
make -f ../scripts/build.mk obj=platform/cmsis/DSP_Lib 
make -f ../scripts/build.mk obj=platform/drivers 
make -f ../scripts/build.mk obj=platform/drivers/ana 
make -f ../scripts/build.mk obj=platform/drivers/ana/best2300p 
make -f ../scripts/build.mk obj=platform/drivers/bt 
make -f ../scripts/build.mk obj=platform/drivers/bt/best2300p 
make -f ../scripts/build.mk obj=platform/drivers/btpcm 
make -f ../scripts/build.mk obj=platform/drivers/codec 
make -f ../scripts/build.mk obj=platform/drivers/codec/best2300p 
make -f ../scripts/build.mk obj=platform/drivers/norflash 
make -f ../scripts/build.mk obj=platform/drivers/sbcacc 
make -f ../scripts/build.mk obj=platform/drivers/usb 
make -f ../scripts/build.mk obj=platform/drivers/usb/usb_dev 
make -f ../scripts/build.mk obj=platform/hal 
make -f ../scripts/build.mk obj=platform/hal/best2300p 
make -f ../scripts/build.mk obj=platform/main 
make -f ../scripts/build.mk obj=platform/main/../../utils/hwtimer_list 
make -f ../scripts/build.mk obj=services 
make -f ../scripts/build.mk obj=services/../utils/crc16 
make -f ../scripts/build.mk obj=services/../utils/crc32 
make -f ../scripts/build.mk obj=services/../utils/heap 
make -f ../scripts/build.mk obj=services/app_ibrt 
make -f ../scripts/build.mk obj=services/app_tws 
make -f ../scripts/build.mk obj=services/audio_dump 
make -f ../scripts/build.mk obj=services/audio_process 
make -f ../scripts/build.mk obj=services/audioflinger 
make -f ../scripts/build.mk obj=services/bt_app 
make -f ../scripts/build.mk obj=services/bt_app/a2dp_codecs 
make -f ../scripts/build.mk obj=services/bt_if_enhanced 
make -f ../scripts/build.mk obj=services/bt_profiles_enhanced 
make -f ../scripts/build.mk obj=services/communication 
make -f ../scripts/build.mk obj=services/cp_accel 
make -f ../scripts/build.mk obj=services/hw_dsp 
make -f ../scripts/build.mk obj=services/ibrt_core 
make -f ../scripts/build.mk obj=services/ibrt_ui 
make -f ../scripts/build.mk obj=services/norflash_api 
make -f ../scripts/build.mk obj=services/nv_section 
make -f ../scripts/build.mk obj=services/nv_section/aud_section 
make -f ../scripts/build.mk obj=services/nv_section/customparam_section 
make -f ../scripts/build.mk obj=services/nv_section/factory_section 
make -f ../scripts/build.mk obj=services/nv_section/log_section 
make -f ../scripts/build.mk obj=services/nv_section/userdata_section 
make -f ../scripts/build.mk obj=services/nvrecord 
make -f ../scripts/build.mk obj=services/osif 
make -f ../scripts/build.mk obj=services/overlay 
make -f ../scripts/build.mk obj=services/resources 
make -f ../scripts/build.mk obj=apps 
make -f ../scripts/build.mk obj=apps/anc 
make -f ../scripts/build.mk obj=apps/audioplayers 
make -f ../scripts/build.mk obj=apps/audioplayers/a2dp_decoder 
make -f ../scripts/build.mk obj=apps/battery 
make -f ../scripts/build.mk obj=apps/common 
make -f ../scripts/build.mk obj=apps/factory 
make -f ../scripts/build.mk obj=apps/key 
make -f ../scripts/build.mk obj=apps/main 
make -f ../scripts/build.mk obj=apps/pwl 
make -f ../scripts/build.mk obj=utils/cqueue 
make -f ../scripts/build.mk obj=utils/list 
make -f ../scripts/build.mk obj=services/multimedia 
make -f ../scripts/build.mk obj=services/multimedia/audio/process/anc/cfg 
make -f ../scripts/build.mk obj=services/multimedia/audio/process/filters/cfg 
make -f ../scripts/build.mk obj=services/multimedia/audio/process/resample/coef 
make -f ../scripts/build.mk obj=utils/intersyshci 
make -f ../scripts/build.mk obj=utils/libc 
make -f ../scripts/build.mk obj=rtos 
make -f ../scripts/build.mk obj=rtos/rtx/TARGET_CORTEX_M 
make -f ../scripts/build.mk obj=utils/crash_catcher 
make -f ../scripts/build.mk obj=utils/xyzmodem 
make -f ../scripts/build.mk obj=config 
make -f ../scripts/build.mk obj=utils/boot_struct 
make[1]: Leaving directory '/home/bryton/src/github.com/hall/little-buddy/out'
MAKE END: 2024-01-20 22:41:13.680109481
MAKE TIME: 1.50 seconds
```

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -4,6 +4,10 @@ This project uses the [nix](https://nixos.org/download.html) package manager.
 
 > **NOTE**: until they're no longer experimental, you'll need to [enable flakes](https://nixos.wiki/wiki/Flakes#Enable_flakes)
 
+## Enable LFS
+
+**IMPORTANT:*** You must enable git LFS; otherwise, the `.a` files included in this repo will be stubs rather than the necessary binaries.
+
 ## build
 
 Enter a development environment with

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -4,11 +4,9 @@ This project uses the [nix](https://nixos.org/download.html) package manager.
 
 > **NOTE**: until they're no longer experimental, you'll need to [enable flakes](https://nixos.wiki/wiki/Flakes#Enable_flakes)
 
-## Enable LFS
-
-**IMPORTANT:*** You must enable git LFS; otherwise, the `.a` files included in this repo will be stubs rather than the necessary binaries.
-
 ## build
+
+> **WARN**: git LFS must be enabled to resolve various `.a` binary files
 
 Enter a development environment with
 

--- a/flake.nix
+++ b/flake.nix
@@ -116,12 +116,13 @@
                 ffmpeg
                 xxd
               ];
-              buildPhase = ''
-                if [ "$(head -c 7 ./platform/drivers/usb/usb_dev/lib/libusbdev.a)" = "version" ] ; then
-                  2>&1 echo "ERROR: Wrong binary contents; please make sure Git LFS is enabled and binaries have been pulled"
+              postUnpack = ''
+                if [ "$(head -c 7 $(git lfs ls-files --name-only | head -1))" =! "version" ]; then
+                  echo "ERROR: run 'git lfs pull' to resolve binary files"
                   exit 1
                 fi
-
+              '';
+              buildPhase = ''
                 find ./config/res/spoken -maxdepth 1 -mindepth 1 -type d -exec sh -c 'make -j LANGUAGE=$(basename {})' \;
               '';
               installPhase = ''

--- a/flake.nix
+++ b/flake.nix
@@ -117,6 +117,11 @@
                 xxd
               ];
               buildPhase = ''
+                if [ "$(head -c 7 ./platform/drivers/usb/usb_dev/lib/libusbdev.a)" = "version" ] ; then
+                  2>&1 echo "ERROR: Wrong binary contents; please make sure Git LFS is enabled and binaries have been pulled"
+                  exit 1
+                fi
+
                 find ./config/res/spoken -maxdepth 1 -mindepth 1 -type d -exec sh -c 'make -j LANGUAGE=$(basename {})' \;
               '';
               installPhase = ''

